beater/beater.go

```diff
@@ -193,12 +195,25 @@ func initTracer(info beat.Info, config *Config, logger *logp.Logger) (*elasticap
 		return nil, nil, err
 	}
 	if config.SelfInstrumentation.isEnabled() {
+		if config.SelfInstrumentation.Hosts != nil {
+			t, err := transport.NewHTTPTransport(config.SelfInstrumentation.Hosts[0], config.SelfInstrumentation.SecretToken)
```

Comment:
probably worth a comment here stating that Hosts will only ever have one element, and that extending beyond that depends on https://github.com/elastic/apm-agent-go/issues/200